src/vstest.console/Processors/ResultsDirectoryArgumentProcessor.cs

```diff
@@ -139,20 +139,14 @@ public void Initialize(string argument)
 
             try
             {
-                // to check valid directory path
-                if (!string.IsNullOrEmpty(argument) && !IsValidPath(argument) )
-                {
-                    throw new ArgumentException("Illegal characters in path.");
-                }
-
-                var di = new DirectoryInfo(argument);
-
                 if (!Path.IsPathRooted(argument))
                 {
                     argument = Path.GetFullPath(argument);
                 }
+
+                var di = Directory.CreateDirectory(argument);
```

Comment:
Just as an FYI.

This will only validate that you have access to this path "at this moment" in time. It could fail here and still be valid later (some step that runs after this check, but before results are actually written could make the path available).

It could also succeed here and fail later (a network share or path could exist when checked and go down before the results are actually written).